lib/rules/summary-interactive-matches.js

```diff
@@ -0,0 +1,44 @@
+import { hasContentVirtual } from '../commons/dom';
+
+export default function summaryIsInteractiveMatches(_, virtualNode) {
+  // Summary only interactive if its real DOM parent is a details element
+  const parent = virtualNode.parent;
+  if (
+    parent.props.nodeName !== 'details' ||
+    virtualNode.shadowId !== parent.shadowId
+  ) {
+    return false;
+  }
+  if (!hasDetailsContent(parent)) {
+    return false;
+  }
+  // Only the first summary element is interactive
+  const firstSummary = parent.children.find(
+    child => child.props.nodeName === 'summary'
+  );
+  if (firstSummary !== virtualNode) {
+    return false;
+  }
+  return true;
+}
+
+function hasDetailsContent(vDetails) {
```

Comment:
Why ignore summary elements of details that don't have content? They are still interactive and screen readers read them out. They don't seem to be treated any differently if the details has text content.

```html
<details>
  <summary></summary>
</details>
```

* JAWS / Chrome - "unlabeled zero button, collapsed"
* NVDA / Firefox - "filled right pointing small triangle, collapsed"
* VO / Safari - "collapsed, details"

Reply:
I'm not less sure what's going on when there's nothing in the details element to expand. Its just an empty tab stop at that point. Those aren't great, but they're not an WCAG violation IMO.